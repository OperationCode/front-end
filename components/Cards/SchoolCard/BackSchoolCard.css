.disclaimer {
  font-size: 0.825rem;
}

.divider {
  border: 1px solid var(--white);
  width: 100%;
}

.locations {
  font-size: 1.25rem;
  list-style: none;
  margin: 0;
  max-height: 150px;
  overflow-y: auto;
  padding: 0;
  width: 85%;
}
