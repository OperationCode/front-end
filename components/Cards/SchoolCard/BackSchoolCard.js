import React from 'react';
import PropTypes from 'prop-types';
import CloseButton from 'components/_common_/CloseButton/CloseButton';
import styles from './BackSchoolCard.css';

BackSchoolCard.propTypes = {
  cardFlipCallback: PropTypes.func.isRequired,
  locations: PropTypes.arrayOf(
    PropTypes.shape({
      address1: PropTypes.string,
      address2: PropTypes.string,
      city: PropTypes.string,
      state: PropTypes.string,
      va_accepted: PropTypes.bool.isRequired,
      zip: PropTypes.number,
    }),
  ).isRequired,
  logoSource: PropTypes.string.isRequired,
  schoolName: PropTypes.string.isRequired,
};

function BackSchoolCard({ cardFlipCallback, locations, logoSource, schoolName }) {
  return (
    <>
      <CloseButton onClick={cardFlipCallback} theme="white" />
      <img src={logoSource} alt={`${schoolName} logo`} />
      <span className={styles.disclaimer}>
        <b>*</b> denotes a location that accepts the GI Bill
      </span>
      <hr className={styles.divider} />
      <ul className={styles.locations}>
        {locations.map(location => (
          <li key={`${location.address1} ${location.address2}`}>
            {location.city && location.state && `${location.city}, ${location.state}`}
            {location.va_accepted ? '*' : null}
          </li>
        ))}
      </ul>
    </>
  );
}

export default BackSchoolCard;
