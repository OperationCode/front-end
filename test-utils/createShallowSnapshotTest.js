import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
