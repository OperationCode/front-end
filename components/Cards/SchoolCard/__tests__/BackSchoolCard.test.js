import React from 'react';
import { mount } from 'enzyme';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
import BackSchoolCard from '../BackSchoolCard';

const locations = [
  {
    address1: '825 Battery Street',
    address2: '3rd Floor',
    city: 'San Francisco',
    state: 'CA',
    va_accepted: true,
    zip: 94111,
  },
  {
    address1: '123 Battery Street',
    address2: 'Ste 1',
    city: 'San Diego',
    state: 'CA',
    va_accepted: false,
    zip: 90111,
  },
];

describe('BackSchoolCard', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(
      <BackSchoolCard
        cardFlipCallback={jest.fn()}
        locations={locations}
        logoSource="logoSource"
        schoolName="alt"
      />,
    );
  });

  it('should call callback function when button is clicked', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(
      <BackSchoolCard
        cardFlipCallback={onClickMock}
        locations={locations}
        logoSource="logoSource"
        schoolName="alt"
      />,
    );
    wrapper.find('button').simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
