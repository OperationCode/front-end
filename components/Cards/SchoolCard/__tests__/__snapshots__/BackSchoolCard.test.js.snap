// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackSchoolCard should render with required props 1`] = `
<React.Fragment>
  <CloseButton
    disabled={false}
    onClick={[MockFunction]}
    theme="white"
  />
  <img
    alt="alt logo"
    src="logoSource"
  />
  <span
    className="disclaimer"
  >
    <b>
      *
    </b>
     denotes a location that accepts the GI Bill
  </span>
  <hr
    className="divider"
  />
  <ul
    className="locations"
  >
    <li>
      San Francisco, CA
      *
    </li>
    <li>
      San Diego, CA
    </li>
  </ul>
</React.Fragment>
`;
